File: Sources/ArgentUtilsCore/PRTarget.swift
import Foundation

/// Whose PRs a wizard acts on: my own, another user's, or one specific PR by
/// number/URL. Shared by the Review and Resolve-conflicts wizards (and mirrored in
/// the Linux front-end's `prtarget.py`).
public enum PRTarget: Int, CaseIterable, Identifiable {
    case mine, someone, specific
    public var id: Int { rawValue }
    public var title: String {
        switch self {
        case .mine:     return "Mine"
        case .someone:  return "Someone else's"
        case .specific: return "Specific PR"
        }
    }
}

File: linux/co_maintainer/prtarget.py
"""Whose PRs a wizard acts on (mirrors ``PRTarget.swift``).

My own, another user's, or one specific PR by number/URL. Shared by the Review and
Resolve-conflicts wizards.
"""

from __future__ import annotations

from enum import IntEnum


class PRTarget(IntEnum):
    MINE = 0
    SOMEONE = 1
    SPECIFIC = 2

    @property
    def title(self) -> str:
        return {
            PRTarget.MINE: "Mine",
            PRTarget.SOMEONE: "Someone else's",
            PRTarget.SPECIFIC: "Specific PR",
        }[self]
